File: server/kotlin/user/src/main/java/xyz/hoper/util/ReflectionUtil.java
package xyz.hoper.util;

import org.reflections.Reflections;
import org.reflections.util.ClasspathHelper;
import org.reflections.util.ConfigurationBuilder;
import org.reflections.util.FilterBuilder;

import java.util.Arrays;
import java.util.stream.Stream;

/**
 * reflections反射工
 */
public class ReflectionUtil {

    public static Reflections getReflections(String... packageAddress) {
        ConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
        Stream.of(packageAddress).forEach(str -> configurationBuilder.addUrls(ClasspathHelper.forPackage(str.trim())));
        FilterBuilder filterBuilder = new FilterBuilder();
        Arrays.stream(packageAddress).forEach(filterBuilder::includePackage);
        configurationBuilder.filterInputsBy(filterBuilder);
        return new Reflections(configurationBuilder);
    }

    public static Reflections getReflections(String packageAddress) {
        ConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
        Stream.of(packageAddress).forEach(str -> configurationBuilder.addUrls(ClasspathHelper.forPackage(str.trim())));
        FilterBuilder filterBuilder = new FilterBuilder();
        filterBuilder.includePackage(packageAddress);
        configurationBuilder.filterInputsBy(filterBuilder);
        return new Reflections(configurationBuilder);
    }

}

File: server/static/kotlin/v2/web/src/main/kotlin/xyz/hoper/utils/common/ReflectionUtil.kt
package xyz.hoper.utils.common

import org.reflections.Reflections
import org.reflections.util.ClasspathHelper
import org.reflections.util.ConfigurationBuilder
import org.reflections.util.FilterBuilder
import java.util.stream.Stream

/**
 * reflections反射工具
 */
object ReflectionUtil {
    fun getReflections(vararg packageAddress: String?): Reflections {
        val configurationBuilder = ConfigurationBuilder()
        val filterBuilder = FilterBuilder()
        packageAddress.forEach { str -> configurationBuilder.addUrls(ClasspathHelper.forPackage(str?.trim { it <= ' ' })) }
        filterBuilder.includePackage(*packageAddress)
        configurationBuilder.filterInputsBy(filterBuilder)
        return Reflections(configurationBuilder)
    }
}